src/null-safety/index.md

```diff
@@ -1,39 +1,247 @@
 ---
 title: Null safety
-description: Watch this space for news about null safety
-toc: false
+description: Information about Dart's upcoming null safety feature
 ---
 
-Watch this page for updates about null safety.
-In the meantime, check out these resources:
+Sound null safety is coming to the Dart language!
+Once you opt into null safety,
+types are non-nullable by default, meaning that
+values in your code can’t be null _unless you say they can be._
+With null safety, your **runtime** null-reference exceptions
+turn into **edit-time** analysis errors.
+The compiler can optimize away internal null checks,
+enabling apps to be faster, smaller, and more reliable.
 
-* [Null-safe playground: nullsafety.dartpad.dev][nullsafety.dartpad.dev]
+You can practice using null safety in the web app
+[DartPad with Null Safety.][nullsafety.dartpad.dev]
+
+{{ site.alert.important }}
+  **Don't start migrating large projects yet.**
+  Do test the feature and [give us feedback.][]
+{{ site.alert.end }}
+
+[give us feedback.]: https://github.com/dart-lang/sdk/issues/new?title=Null%20safety%20feedback:%20[issue%20summary]&labels=NNBD&body=Describe%20the%20issue%20or%20potential%20improvement%20in%20detail%20here
+
+If you've used Kotlin, TypeScript, or C#,
+then some of the null-safety syntax might look familiar.
+That's by design: the Dart language aims to be unsurprising.
+
+
+## Creating and using variables (?, !, late)
+
+Use `?`, `!`, and `late` to tweak how the Dart analyzer treats your code.
+
+Here’s an example of declaring a **non-nullable integer variable**
+(assuming you’ve opted into null safety):
+
+```dart
+int aRealInt; // Can't be used until it's initialized to an integer.
+```
+
+If the variable _can_ have the value `null`,
+**add  `?`** to its type declaration:
+
+```dart
+int? aNullableInt = null;
+```
+
+When using a value that you know isn’t null but that has a nullable type,
+**add `!`** to tell the Dart analyzer that the value isn’t null:
+
+```dart
+int? a = 2;
+int b = a!; // The ! temporarily changes `a` from int? to int.
+```
+
+If you know that a non-nullable value will be
+initialized to a non-null value before it's used,
+but the Dart analyzer thinks otherwise,
+**add the `late` keyword** to the type:
+
+```dart
+late int aRealInt;
+```
+
+{% comment %}
+{{ site.alert.tip }}
+  Avoid using `late final` unless you really need it.
+  For example, don't use `late final` in functions,
+  and in constructors prefer using an [initializer list][].
```

Comment:
Constructors don't declare late final variables, so this is referring to instance variables?

If so, then maybe "in classes, prefer fields initialized using an [initializer list][]."